src/compute/block.rs

```diff
@@ -0,0 +1,655 @@
+
+    // If both min and max in a given axis are set and max <= min then this determines the size in that axis
+    let min_max_definite_size = min_size.zip_map(max_size, |min, max| match (min, max) {
+        (Some(min), Some(max)) if max <= min => Some(min),
+        _ => None,
+    });
+
+    // Block nodes automatically stretch fit their width to fit available space if available space is definite
+    let available_space_based_size =
+        Size { width: available_space.width.into_option().maybe_sub(margin.horizontal_axis_sum()), height: None };
+
+    let styled_based_known_dimensions =
+        known_dimensions.or(min_max_definite_size).or(clamped_style_size).or(available_space_based_size);
+
+    // Short-circuit layout if the container's size is fully determined by the container's size and the run mode
+    // is ComputeSize (and thus the container's size is all that we're interested in)
+    if run_mode == RunMode::ComputeSize {
+        if let Size { width: Some(width), height: Some(height) } = styled_based_known_dimensions {
+            return Size { width, height }.into();
+        }
+    }
+
+    #[cfg(feature = "debug")]
+    NODE_LOGGER.log("BLOCK");
+    compute_inner(
+        tree,
+        node_id,
+        styled_based_known_dimensions,
+        parent_size,
+        available_space,
+        run_mode,
+        vertical_margins_are_collapsible,
+    )
+}
+
+/// Computes the layout of [`LayoutTree`] according to the block layout algorithm
+fn compute_inner(
+    tree: &mut impl LayoutTree,
+    node_id: NodeId,
+    known_dimensions: Size<Option<f32>>,
+    parent_size: Size<Option<f32>>,
+    available_space: Size<AvailableSpace>,
+    run_mode: RunMode,
+    vertical_margins_are_collapsible: Line<bool>,
+) -> SizeBaselinesAndMargins {
+    let style = tree.style(node_id);
+    let raw_padding = style.padding;
+    let raw_border = style.border;
+    let raw_margin = style.margin;
+    let aspect_ratio = style.aspect_ratio;
+    let size = style.size.maybe_resolve(parent_size).maybe_apply_aspect_ratio(aspect_ratio);
+    let min_size = style.min_size.maybe_resolve(parent_size).maybe_apply_aspect_ratio(aspect_ratio);
+    let max_size = style.max_size.maybe_resolve(parent_size).maybe_apply_aspect_ratio(aspect_ratio);
+    let padding = style.padding.resolve_or_zero(parent_size.width);
+    let border = style.border.resolve_or_zero(parent_size.width);
+
+    // Scrollbar gutters are reserved when the `overflow` property is set to `Overflow::Scroll`.
+    // However, the axis are switched (transposed) because a node that scrolls vertically needs
+    // *horizontal* space to be reserved for a scrollbar
+    let scrollbar_gutter = {
+        let offsets = style.overflow.transpose().map(|overflow| match overflow {
+            Overflow::Scroll => style.scrollbar_width,
+            _ => 0.0,
+        });
+        // TODO: make side configurable based on the `direction` property
```

Comment:
Maybe we can create an issue for this so we don't forget

Reply:
There is one! https://github.com/DioxusLabs/taffy/issues/213